### ISSUE TITLE
GenerateMips on sRGB Textures interpolating in gamma space?

### ISSUE THREAD

**Issue by StephenMLucas** [microsoft/DirectXTK12#142]
Hi,

Reviewing the implementation of GenerateMips for sRGB textures it looks like the implementation is doing a reinterpret copy into a linear texture format and then using a bilinear sampler in the ComputeShader to perform the average. 

Unless I'm missing something about the math for this case it seems like doing the linear arithmetic on gamma encoded values in the sampler is going to result in color errors in the output mips, as demonstrated by this GLTF test case; https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureLinearInterpolationTest

**Comment by walbourn** [comment]
Right. I should add a inverse gamma/regamma in this case to the sahder...

**Comment by StephenMLucas** [comment]
Presumably the other option would be to use a regular pixel shader pipeline, like the post processing functions, where the sRGB formats are directly supported as both SRVs and RenderTargets to move the conversion before the filtering in the sampler? 

I don't know whether having the compute shader version do 4 point samples, gamma converting, taking the average and regammaing would result in any overall performance difference compared with using the built-in pixel shader functions though.

**Comment by walbourn** [comment]
GenerateMips has always been a bug-farm, and on Direct3D 11 the implementation was very driver-dependent. It does seem like the only "correct" way to solve this is to do as you say and create yet another generatemips codepath using classic texture down-sizing on the GPU.

**Comment by walbourn** [comment]
MiniEngine does it with the compute shader, so I'll crib that.

https://github.com/microsoft/DirectX-Graphics-Samples/blob/master/MiniEngine/Core/Shaders/GenerateMipsCS.hlsli
